### PR TITLE
Remove `fn`/closure `#[spirv(unroll_loops)]` attribute.

### PR DESCRIPTION
**Pull request by eddyb**
This attribute was pretty hacky, as it affected *all* loops in the function body, because MIR is a control-flow graph so we can't see "loop expressions" to get attributes from (see below for how we could rectify that!).

To quote the `CHANGELOG` entry:
> - Removed the `fn`/closure `#[spirv(unroll_loops)]` attribute, as it has no users,
  is becoming non-trivial to support, and requires redesign for better ergonomics
  (e.g. `#[spirv(unroll)]` applied to individual loops, not the whole `fn`/closure)

For more context, see:
* https://github.com/EmbarkStudios/rust-gpu/pull/940#issuecomment-1320004784
 (specifically, supporting it on SPIR-T would require additional effort, but there's no users we can find)
* #944
 (specifically, the ability to handle loop annotations at a per-loop level, would be massively superior)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+<!-- NOTE(eddyb) sections from the original template:
+
+### Added
```
Comment (oisyn) [inline CHANGELOG.md:10]:
```suggestion
### Added ⭐
```

Sorry, I like my emojis 😁

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+- New features go here in a bullet list
+
+### Changed
```
Comment (oisyn) [inline CHANGELOG.md:13]:
```suggestion
### Changed 🛠️
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+- Changes to existing functionality go here in a bullet list
+
+### Deprecated
```
Comment (oisyn) [inline CHANGELOG.md:16]:
```suggestion
### Deprecated 🚧
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+- Mark features soon-to-be removed in a bullet list
+
+### Removed
```
Comment (oisyn) [inline CHANGELOG.md:19]:
```suggestion
### Removed 🔥
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+- Features that have been removed in a bullet list
+
+### Fixed
```
Comment (oisyn) [inline CHANGELOG.md:22]:
```suggestion
### Fixed 🩹
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+- Bug fixes in a bullet list
+
+### Security
```
Comment (oisyn) [inline CHANGELOG.md:25]:
```suggestion
### Security 🔐
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,12 +5,40 @@ All notable changes to this project will be documented in this file.
...
+
+-->
+
 ## [Unreleased]
 
 ### Changed 🛠️
 
 - Applied workspace inheritance to Cargo.toml files
 
+### Removed
```
Comment (oisyn) [inline CHANGELOG.md:36]:
```suggestion
### Removed 🔥
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -29,6 +57,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - 🚨BREAKING🚨 Migrated from `register_attr` to `register_tool`. [More information](docs/src/migration-to-register-tool.md).
 - Updated toolchain to `nightly-2022-10-01`
 - Updated `glam` to `0.22`
+
+### Removed
```
Comment (oisyn) [inline CHANGELOG.md:61]:
```suggestion
### Removed 🔥
```

Good point though 👍

**Review by oisyn — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (oisyn) [inline CHANGELOG.md:25]:
```suggestion
### Added ⭐
- New features go here in a bullet list

### Changed 🛠
- Changes to existing functionality go here in a bullet list

### Deprecated 🚧
- Mark features soon-to-be removed in a bullet list

### Removed 🔥
- Features that have been removed in a bullet list

### Fixed 🩹
- Bug fixes in a bullet list

### Security 🔐
```

Sorry, I like my emojis 😁

File: CHANGELOG.md (hunk 7 above)
Comment (oisyn) [inline CHANGELOG.md:36]:
```suggestion
### Removed 🔥
```

**Review by oisyn — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (eddyb) [reply CHANGELOG.md:25]: Heh, thanks! (I just didn't know where to get the emoji versions from)

**Review by oisyn — APPROVED** [review]

**Pull request merged**